tests/release_workflow.py

```diff
@@ -1,46 +1,43 @@
 """
 Script to update citation.cff file and validate it then draft a release note.
 """
-import argparse
 from datetime import datetime
 
-def read_from_file(infile):
-    with open(infile, "r", encoding="utf8") as fid:
-        return fid.readlines()
+from ruamel.yaml import YAML
+
+# Configure the YAML parser
+yaml = YAML()
+yaml.indent(mapping=3, sequence=2, offset=0)
+yaml.allow_duplicate_keys = True
+yaml.explicit_start = False
 
-def int_to_version(major, minor, tiny):
-    return f"{major}.{minor}.{tiny}"
 
 def update_citation_file(citation_file_path):
-    file_content = read_from_file(citation_file_path)
-    for idx, line in enumerate(file_content):
-        word_array = line.split(" ")
-        if ("version:" in word_array):
-            major, minor, tiny = map(int, word_array[1].split("."))
+    with open(citation_file_path) as stream:
+        file_content = yaml.load(stream)
+
+    major, minor, tiny = file_content["version"].split(".")
+    major, minor, tiny = int(major), int(minor), int(tiny)
 
-            tiny = tiny + 1
-            if (tiny > 10 and minor < 10):
-                tiny = 0
-                minor = minor + 1
-            elif (tiny > 10 and minor > 10):
-                minor = 0
-                major = major + 1
+    # Maybe we want to provide an input to determine what the release version
+    # should be? Here we are restricting the patch and minor versions to be less
+    # than or equal to 10, but there's nothing in semver practices that enforces this.
```

Comment:
Good idea! My current implementation is based on my discussions with Malvika. 

If we change this to take an input, how should we implement it as a variable input for the CLI execution in Github workflows since it will be dependent on how we classify changes in the TW?